xmtp_id/src/associations/signature.rs

```diff
@@ -60,181 +102,253 @@ impl Clone for Box<dyn Signature> {
     }
 }
 
-#[allow(dead_code)]
 #[derive(Clone)]
 pub struct RecoverableEcdsaSignature {
-    signature_text: String,
-    signature_bytes: Vec<u8>,
+    message: String,
+    public_key: [u8; 64],
+    bytes: [u8; 65],
 }
 
 impl RecoverableEcdsaSignature {
-    pub fn new(signature_text: String, signature_bytes: Vec<u8>) -> Self {
-        RecoverableEcdsaSignature {
-            signature_text,
-            signature_bytes,
-        }
+    pub fn new(
+        message: String,
+        public_key: [u8; 64],
+        bytes: [u8; 65],
+    ) -> Result<Self, SignatureError> {
+        let sig = RecoverableEcdsaSignature {
+            message,
+            public_key,
+            bytes,
+        };
+        sig.verify()?;
+        Ok(sig)
     }
 }
 
 impl Signature for RecoverableEcdsaSignature {
-    fn recover_signer(&self) -> Result<MemberIdentifier, SignatureError> {
-        todo!()
+    fn get_kind(&self) -> &SignatureKind {
+        &SignatureKind::Erc191
     }
 
-    fn signature_kind(&self) -> SignatureKind {
-        SignatureKind::Erc191
+    fn get_signer(&self) -> MemberIdentifier {
+        MemberIdentifier::Address(h160addr_to_string(public_key_to_address(self.public_key)))
     }
 
-    fn bytes(&self) -> Vec<u8> {
-        self.signature_bytes.clone()
+    fn to_bytes(&self) -> Vec<u8> {
+        self.bytes.to_vec()
     }
 
     fn to_proto(&self) -> SignatureProto {
         SignatureProto {
             signature: Some(SignatureKindProto::Erc191(RecoverableEcdsaSignatureProto {
-                bytes: self.bytes(),
+                message: self.message.clone(),
```

Comment:
Same as the comment below. The `message` cannot be trusted to come from the payload. It needs to be generated by the client doing the verifying.